### PR TITLE
adapt workflow and build script in order to deploy on maven central

### PR DESCRIPTION
**Pull request by pstaeck**
The purpose of the pull request is to switch deployment of t2iapi (java part) from github packages to maven central. 

# Checklist

The following aspects have been respected by the author of this pull request, confirmed by both pull request assignee **and** reviewer:

* Changelog update (necessity checked and entry added or not added respectively)
  * [X] Pull Request Assignee
  * [x] Reviewer
* README update (necessity checked and entry added or not added respectively)
  * [X] Pull Request Assignee
  * [x] Reviewer

### CODE REVIEW

File: tests/java/build.gradle.kts [hunk 1]
```diff
@@ -14,10 +14,9 @@ val configFileMap = configFile.associate { it.split("=")[0] to it.split("=")[1]
 
 val grpcVersion = configFileMap["JAVA_GRPC_VERSION"]
 val baseVersion = configFileMap["BASE_PACKAGE_VERSION"]!!
-val buildId: String? = System.getenv("GITHUB_RUN_NUMBER")
 val t2iapiVersion: String = when (System.getenv("RELEASE_VERSION") == "1") {
     true -> baseVersion
-    false -> baseVersion + ( buildId?.let { ".$it" } ?: "" )
+    false -> baseVersion + "-SNAPSHOT"
```
Comment (ldeichmann) [inline tests/java/build.gradle.kts:19]: I think we want to keep the buildId for Snapshots, consider doing something like
```suggestion
    false -> baseVersion + ( buildId?.let { ".$it" } ?: "" ) + "-SNAPSHOT"
```

File: .github/workflows/build-java.yml [hunk 2]
```diff
@@ -78,24 +78,19 @@ jobs:
...
           name: test_results
           path: tests/java/build/reports/tests/test/**/*.*
-  deploy-github-packages:
+  deploy-maven-central:
     runs-on: ubuntu-latest
-    needs: run-tests-java11
+    needs: run-tests-java17
     if: github.ref == 'refs/heads/main' || startsWith(github.ref, 'refs/tags/')
     permissions:
       packages: write
```
[the remark this answers is not in the document]
Comment (pstaeck) [reply .github/workflows/build-java.yml:86]: done - remove permissions

File: java/build.gradle.kts [hunk 3]
```diff
@@ -86,12 +112,27 @@ publishing {
 
     repositories {
         maven {
-            name = "GitHubPackages"
-            url = uri("https://maven.pkg.github.com/Draegerwerk/t2iapi")
+            name = "SonatypeStaging"
```
[the remark this answers is not in the document]
Comment (pstaeck) [reply java/build.gradle.kts:115]: done - rename it to "Sonatype"

**Comment by ldeichmann** [comment]
@CodiumAI-Agent /review

**Review by ldeichmann — APPROVED** [review]

**Pull request merged**